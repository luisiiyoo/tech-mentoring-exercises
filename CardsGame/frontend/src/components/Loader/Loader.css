@import url(https://fonts.googleapis.com/css?family=Lato);

.Loader {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  color: var(--font-color-app);
  letter-spacing: 1rem;
  font-size: 120%;
  font-family: Lato, sans-serif;
  text-transform: uppercase;
}

.Loader-Span {
  animation: blur 1.5s alternate infinite;
}

@keyframes blur {
  to {
    filter: blur(5px);
  }
}
