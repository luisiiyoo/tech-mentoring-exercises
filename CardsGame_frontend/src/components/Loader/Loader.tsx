import React, { useEffect } from 'react';
import './Loader.css';

const Loader: React.FC = () => {
  useEffect(() => {
    const loading = document.querySelector('.Loader') as Element;
    if (loading !== null) {
      const loadingMsg = loading.textContent || '';
      const letters = loadingMsg.split('');
      loading.textContent = '';
      letters.forEach((letter, i) => {
        const span = document.createElement('span');
        span.className = 'Loader-Span';
        span.textContent = letter;
        span.style.animationDelay = `${i / loadingMsg.length}s`;
        loading.append(span);
      });
    }
  }, []);
  return <div className="Loader">Loading...</div>;
};

export default Loader;
