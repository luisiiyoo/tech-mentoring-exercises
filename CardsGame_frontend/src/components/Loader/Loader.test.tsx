import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Loader from '../Loader';

describe('Loader', () => {
  afterEach(cleanup);

  it('should render Loading component with Loading message', () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
