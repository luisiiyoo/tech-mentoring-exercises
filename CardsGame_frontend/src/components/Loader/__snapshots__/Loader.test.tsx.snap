// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render Loading component with Loading message 1`] = `
<DocumentFragment>
  <div
    class="Loader"
  >
    <span
      class="Loader-Span"
      style="animation-delay: 0s;"
    >
      L
    </span>
    <span
      class="Loader-Span"
      style="animation-delay: 0.1s;"
    >
      o
    </span>
    <span
      class="Loader-Span"
      style="animation-delay: 0.2s;"
    >
      a
    </span>
    <span
      class="Loader-Span"
      style="animation-delay: 0.3s;"
    >
      d
    </span>
    <span
      class="Loader-Span"
      style="animation-delay: 0.4s;"
    >
      i
    </span>
    <span
      class="Loader-Span"
      style="animation-delay: 0.5s;"
    >
      n
    </span>
    <span
      class="Loader-Span"
      style="animation-delay: 0.6s;"
    >
      g
    </span>
    <span
      class="Loader-Span"
      style="animation-delay: 0.7s;"
    >
      .
    </span>
    <span
      class="Loader-Span"
      style="animation-delay: 0.8s;"
    >
      .
    </span>
    <span
      class="Loader-Span"
      style="animation-delay: 0.9s;"
    >
      .
    </span>
  </div>
</DocumentFragment>
`;
